### ISSUE TITLE
Add CUDA warmup inference on startup

### ISSUE THREAD

**Issue by wpflueger** [wpflueger/ParaKey#31]
## Problem

After model loading in \`model.py\`, no warmup inference is performed. The first real transcription is likely slower due to CUDA kernel compilation, memory allocation patterns, and JIT optimization.

## Proposed Solution

After \`self._model.eval()\` in \`ModelLoader.load()\`, run a short dummy transcription (e.g., 1 second of silence) to warm up the CUDA kernels:

\`\`\`python
# Warmup inference
import numpy as np
warmup_audio = np.zeros(sample_rate, dtype="float32")  # 1 second of silence
with torch.no_grad():
    self._model.transcribe(audio=[torch.from_numpy(warmup_audio)])
logger.info("CUDA warmup complete")
\`\`\`

## Files

- \`backend/src/parakey_backend/model.py\` — \`ModelLoader.load()\`

## Priority

Low — improves first-transcription latency, adds ~1-2s to startup.
